### PR TITLE
refactor: Added new props to GetMetricQueryRange to control Step from without global time range.

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
close https://github.com/SigNoz/signoz/issues/2931

### CODE REVIEW

File: frontend/src/store/actions/dashboard/getQueryResults.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export async function GetMetricQueryRange({
 	tableParams,
 	variables = {},
 	params = {},
+	isStepIntervalSelected = false,
```
Comment (palashgdev) [inline frontend/src/store/actions/dashboard/getQueryResults.ts:29]: use from params

File: frontend/src/container/FormAlertRules/index.tsx [hunk 2]
```diff
@@ -351,6 +353,23 @@ function FormAlertRules({
 		<BasicInfo alertDef={alertDef} setAlertDef={setAlertDef} />
 	);
 
+	const updatedStagedQuery = useMemo((): Query | null => {
```
Comment (palashgdev) [inline frontend/src/container/FormAlertRules/index.tsx:356]: util + test case

File: frontend/src/container/FormAlertRules/ChartPreview/index.tsx [hunk 3]
```diff
@@ -23,6 +23,7 @@ export interface ChartPreviewProps {
 	headline?: JSX.Element;
 	threshold?: number | undefined;
 	userQueryKey?: string;
+	isStepIntervalSelected?: boolean;
```
Comment (palashgdev) [inline frontend/src/container/FormAlertRules/ChartPreview/index.tsx:26]: can we have a good name please according to the component used + having some context around it 😅

File: frontend/src/container/FormAlertRules/index.tsx (hunk 2 above)
Comment (Rajat-Dabade) [reply frontend/src/container/FormAlertRules/index.tsx:356]: Done.

File: frontend/src/store/actions/dashboard/getQueryResults.ts (hunk 1 above)
Comment (Rajat-Dabade) [reply frontend/src/store/actions/dashboard/getQueryResults.ts:29]: Okay

File: frontend/src/container/FormAlertRules/ChartPreview/index.tsx (hunk 3 above)
Comment (Rajat-Dabade) [reply frontend/src/container/FormAlertRules/ChartPreview/index.tsx:26]: Okay.

**Comment by srikanthccv** [comment]
@palashgdev Please take a look again. I see the comments are addressed. If a bug is fixed, there is no good reason to delay it. @Rajat-Dabade, please resolve conflicts.

**Comment by srikanthccv** [comment]
@Rajat-Dabade, please resolve conflicts. @palashgdev, please take some time to go through the backlog PRs and review them.

File: frontend/src/store/actions/dashboard/getQueryResults.ts [hunk 4]
```diff
@@ -21,6 +21,18 @@ export async function GetMetricQueryRange(
 
 	const response = await getMetricsQueryRange(queryPayload);
 
+	const response = await getMetricsQueryRange({
```
[the remark this answers is not in the document]
Comment (Rajat-Dabade) [reply frontend/src/store/actions/dashboard/getQueryResults.ts:24]: @palashgdev, It is already removed, don't have any idea why it is showing like this. 
I have checked out the code and confirmed.

[the remark this answers is not in the document]
Comment (palashgdev) [reply frontend/src/store/actions/dashboard/getQueryResults.ts:24]: if it is removed then why it is showing this as an addition

**Comment by palashgdev** [comment]
this is not even required why we are sending it here ?

Screenshot 2023-10-04 at 3 33 57 PM

**Review by palashgdev — DISMISSED** [review]

**Comment by palashgdev** [comment]
@makeavish can you please verify

**Comment by makeavish** [comment]
@Rajat-Dabade Please resolve conflict

**Review by palashgdev — APPROVED** [review]

**Pull request merged**